.github/workflows/deploy-server.yml

```
@@ -15,11 +15,13 @@ jobs:
       infra_changed: ${{ steps.filter.outputs.infra }}
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetch full history to allow comparing against previous commits
       - name: Detect changed areas
         id: filter
         uses: dorny/paths-filter@v2
```

actionlint (1.7.8)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)